### ISSUE TITLE
"Move pulled object" keybind / PullController should be using entity coordinates instead of map coordinates

### ISSUE THREAD

**Issue by ElectroJr** [space-wizards/space-station-14#4573]
The "Move pulled object" (ctrl+right-click) keybinding is currenly partially broken, depending on where you are on the map.

`Content.Server.Physics.Controllers.PullController` currently compares the `MovingTo` map coordinates with a pullable entity's current map coordinates using `Vector2.EqualsApprox`. This checks if both of the individual vector components are **relatively equal**. As a result it considers the map coordinates: (500, 0) and (504.9, 0) to be equal, even though they are ~5 tiles apart. The effect is that when not near the centre of the map, trying to move a pulled entity using ctrl+right-click considers relatively large distances to be the "same" position, effectively stopping you from using the "Move pulled object" action.

Instead, this should probably either use vectors defined in the pulled entity's reference frame, or use an absolute distance measure. If using the former solution, then the `PullController.MaximumSettleDistance` variable should probably be renamed to make it clear that it is a tolerance measure, not an absolute distance. 

On a related note, `Robust.Shared.Maths.MathHelper.CloseTo` does not behave as described in it's comments. It uses the largest values do determine the tolerance level, not the smallest one.

**Comment by Zumorica** [comment]
> As a result it considers the map coordinates: (500, 0) and (504.9, 0) to be equal, even though they are ~5 tiles apart. 
How is this even possible? I'm looking at the code and everything seems fine

**Comment by ElectroJr** [comment]
Approx equals checks if the X and Y components are approximately equal using
`MathHelper.CloseTo(X, other.X, tolerance) && MathHelper.CloseTo(Y, other.Y, tolerance);`
with `tolerance = MaximumSettleDistance`  (i.e., 0.01)

MathHelper returns **true** if the difference between the two values is less than EITHER the tolerance, or the largest value multiplied by the tolerance (contradicting it's own comments).

So in my example, 0==0 is trivially approximately equal. Meanwhile 500 and 504.9 are within 1% of each other, so for the given tolerance it considers the X coordinates equal.

For example, just try running:
Basically just try running:
```cs
var vec1 = new Vector2(504.9f, 0f);
var vec2 = new Vector2(500f, 0f);
var equals = vec1.EqualsApprox(vec2, tolerance: 0.01);
```

**Comment by ElectroJr** [comment]
fixed by #4708
